engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/ordering/LessThanNode.java

```diff
@@ -0,0 +1,305 @@
+package org.enso.interpreter.node.expression.builtin.ordering;
+
+import com.ibm.icu.text.Normalizer;
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import org.enso.interpreter.dsl.AcceptsError;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.error.WarningsLibrary;
+import org.enso.interpreter.runtime.number.EnsoBigInteger;
+
+@BuiltinMethod(
+    type = "Comparable",
+    name = "less_than_builtin",
+    description = """
+        Returns true if self is less than `other`. Or throw an error if the values are
```

Comment:
Huge node with a lot of specialization. I hope we are not going to copy it for `<=`, `>`, `>=`...

Reply:
Definitely not:
- `x > y` is equivalent to `Comparable.less_than_builtin y x` (i.e. `y < x`). 
- `x <= y` is equivalent to `Comparable.less_than_builtin x y || Comparable.equals_builtin x y`.
- `x < y` is equivalent to `Comparable.less_than_builtin x y`.
- `x <= y` is equivalent to `Comparable.less_than_builtin x y || Comparable.equals_builtin x y`.

So `equals_builtin` and `less_than_builtin` are the only builtin nodes that we need for all the comparison operators.

Reply:
What about `compare` builtin returning `-1`, '0' or `1`? That could share the code between `LessThanNode` and `EqualsNode`. Just an idea - who knows what effect it would have on speed - and duplication is kind of OK (better than triplication, etc.).

Reply:
There is little duplication between `LessThanNode` and `EqualsNode`. They both have very different semantics. Firstly, `LessThanNode` handles just a small subset of values as opposed to `EqualsNode`. Second of all, `LessThanNode` fails with `Type_Error` as fallback, whereas `EqualsNode` just returns false as a fallback.

Reply:
In Enso itself our 'primitive' for comparing values is `compare_to`, right? And the operators like `<` etc. are derived from that `compare_to`.
<img width="371" alt="image" src="https://user-images.githubusercontent.com/1436948/217785808-2bf81d62-d476-4487-8944-249f3e61c7d0.png">

(with the exception for numeric types where the operators `>` were builtin for performance)

Shouldn't we keep that convention? `compare_to` seems more semantically clear and performant (because stuff like `<=` will need to call `<` and `==` - so two calls, and the `<` relies on the `comparator` returning an ordering anyway, right? So it seems that basing this on `compare_to` will also be just more _direct_)